src/BindPattern.jl

```diff
@@ -278,3 +281,22 @@ function bind_pattern!(
 
     return (pattern, assigned)
 end
+
+# (interpolation, assigned0) = subst_patvars(interpolation, assigned)
+function subst_patvars(expr, assigned::ImmutableDict{Symbol, Symbol})
+    new_assigned = ImmutableDict{Symbol, Symbol}()
+    # postwalk(f, x) = walk(x, x -> postwalk(f, x), f)
+    new_expr = MacroTools.postwalk(expr) do patvar
+        if patvar isa Symbol
+            tmpvar = get(assigned, patvar, nothing)
+            if tmpvar isa Symbol
+                if !(patvar in keys(new_assigned))
+                    new_assigned = ImmutableDict{Symbol, Symbol}(new_assigned, patvar, tmpvar)
+                end
+                return :($identity($tmpvar))
+            end
+        end
+        patvar
+    end
+    (new_expr, new_assigned)
+end
```

Comment:
Just for the future (not for this PR) :  It might be interesting to look at the MuttsDict implementation - https://github.com/RelationalAI-oss/MuttsDicts.jl .